orchestration/src/main/java/com/sap/ai/sdk/orchestration/spring/OrchestrationChatOptions.java

```diff
@@ -0,0 +1,224 @@
+  }
+
+  private void setPresencePenalty(@Nonnull final Double presencePenalty) {
+    setLlmConfigParam(PRESENCE_PENALTY.getName(), presencePenalty);
+  }
+
+  /**
+   * Returns the stop sequences to use for the chat.
+   *
+   * @return the stop sequences to use for the chat
+   */
+  @Nullable
+  @Override
+  public List<String> getStopSequences() {
+    return getLlmConfigParam("stop_sequences");
+  }
+
+  private void setStopSequences(@Nonnull final List<String> stopSequences) {
+    setLlmConfigParam("stop_sequences", stopSequences);
+  }
+
+  /**
+   * Returns the temperature to use for the chat.
+   *
+   * @return the temperature to use for the chat
+   */
+  @Nullable
+  @Override
+  public Double getTemperature() {
+    return getLlmConfigParam(TEMPERATURE.getName());
+  }
+
+  private void setTemperature(@Nonnull final Double temperature) {
+    setLlmConfigParam(TEMPERATURE.getName(), temperature);
+  }
+
+  /**
+   * Returns the top K to use for the chat.
+   *
+   * @return the top K to use for the chat
+   */
+  @Nullable
+  @Override
+  public Integer getTopK() {
+    return getLlmConfigParam("top_k");
+  }
+
+  private void setTopK(@Nonnull final Integer topK) {
+    setLlmConfigParam("top_k", topK);
+  }
+
+  /**
+   * Returns the top P to use for the chat.
+   *
+   * @return the top P to use for the chat
+   */
+  @Nullable
+  @Override
+  public Double getTopP() {
+    return getLlmConfigParam(TOP_P.getName());
+  }
+
+  private void setTopP(@Nonnull final Double topP) {
+    setLlmConfigParam(TOP_P.getName(), topP);
+  }
+
+  /**
+   * Returns a copy of this {@link OrchestrationChatOptions}.
+   *
+   * @return a copy of this {@link OrchestrationChatOptions}
+   */
+  @SuppressWarnings("unchecked") // The same suppress is in DefaultChatOptions
+  @Nonnull
+  @Override
+  public <T extends ChatOptions> T copy() {
+    val copy = new OrchestrationChatOptions(config);
+    copy.setModel(this.getModel());
+    copy.setModelVersion(this.getModelVersion());
+    if (getFrequencyPenalty() != null) {
+      copy.setFrequencyPenalty(this.getFrequencyPenalty());
+    }
+    if (getMaxTokens() != null) {
+      copy.setMaxTokens(this.getMaxTokens());
+    }
+    if (getPresencePenalty() != null) {
+      copy.setPresencePenalty(this.getPresencePenalty());
+    }
+    if (getStopSequences() != null) {
+      copy.setStopSequences(new ArrayList<>(this.getStopSequences()));
+    }
+    if (getTemperature() != null) {
+      copy.setTemperature(this.getTemperature());
+    }
+    if (getTopK() != null) {
+      copy.setTopK(this.getTopK());
+    }
+    if (getTopP() != null) {
+      copy.setTopP(this.getTopP());
+    }
+    return (T) copy;
+  }
+
+  @SuppressWarnings("unchecked")
+  @Nullable
+  private <T> T getLlmConfigParam(@Nonnull final String param) {
+    if (getLlmConfigNonNull().getModelParams() instanceof LinkedHashMap) {
+      return ((LinkedHashMap<String, T>) getLlmConfigNonNull().getModelParams()).get(param);
```

Comment:
`T` could be `Parameter<ValueT>` (from the class `OrchestrationAiModel`)
But I didn't find a way to do this...

Reply:
I tested that:

```java
private <T> Option<T> getLlmConfigParam(@Nonnull final String param) {
  return Option.of(getLlmConfigNonNull().getModelParams())
    .filter(p -> p instanceof Map<?,?>)
    .map(p -> (T) ((Map<?,?>) p).get(param))
  .filter(Objects::nonNull);
}
```

Unfortunately upon usage, the impliciit `T` is not recognized:

```diff
  public Double getPresencePenalty() {
-   return getLlmConfigParam(PRESENCE_PENALTY.getName());
+   return getLlmConfigParam(PRESENCE_PENALTY.getName()).getOrNull();
  }
```



However this would work, but it looks worse:
```diff
  public Double getPresencePenalty() {
-   return getLlmConfigParam(PRESENCE_PENALTY.getName());
+   return this.<Double>getLlmConfigParam(PRESENCE_PENALTY.getName()).getOrNull();
  }
```

Reply:
I like `this.<Double>getLlmConfigParam(PRESENCE_PENALTY)`. Will implement it